python/lsst/ts/genericcamera/driver/alliedvisioncamera.py

```diff
@@ -0,0 +1,302 @@
+            with self.camera:
+                self.camera.AcquisitionMode.set("SingleFrame")
+                self.camera.ExposureAuto.set("Continuous")
+                self.camera.set_pixel_format(vimba.PixelFormat.Mono8)
+        super().start_live_view()
+
+    def stop_live_view(self):
+        """Stops an active live view data stream from the camera.
+
+        This should review the image format back to original format."""
+        self.is_live_exposure = False
+        with vimba.Vimba.get_instance():
+            with self.camera:
+                self.camera.set_pixel_format(self.normal_image_type)
+        super().stop_live_view()
+
+    def _set_camera_exposure_time(self):
+        """Set the exposure time on the camera."""
+        with vimba.Vimba.get_instance():
+            with self.camera:
+                if not self.is_live_exposure or not self.liveview_use_autoexposure:
+                    self.log.debug("Setting camera exposure time")
+                    self.camera.ExposureAuto.set("Off")
+                    self.camera.ExposureTimeAbs.set(self.exposure_time * MICROSECONDS)
+
+    async def start_take_image(self, exp_time, shutter, science, guide, wfs):
+        """Start taking an image or a set of images.
+
+        Parameters
+        ----------
+        exp_time : float
+            The exposure time in seconds.
+        shutter : bool
+            Should the shutter be opened?
+        science : bool
+            Should the science/main sensor be used?
+        guide : bool
+            Should guider sensor be used?
+        wfs : bool
+            Should wave front sensor be used?
+        """
+        self.exposure_time = exp_time
+        await self.loop.run_in_executor(self.executor, self._set_camera_exposure_time)
+        await super().start_take_image(exp_time, shutter, science, guide, wfs)
+
+    def _get_frame(self):
+        """Wait for the frame to arrive from the camera.
+
+        Returns
+        -------
+        Vimba.Frame
+            The frame captured from the camera.
+        """
+        with vimba.Vimba.get_instance():
+            # v.enable_log(vimba.LOG_CONFIG_INFO)
+            with self.camera:
+                if self.liveview_use_autoexposure:
+                    timeout = int(
+                        (self.camera.ExposureTimeAbs.get() / MILLISECONDS)
+                        + 2 * MILLISECONDS
+                    )
+                else:
+                    timeout = int(self.exposure_time + 2) * MILLISECONDS
+                self.log.debug(f"Exposure timeout = {timeout} ms")
+                frame = self.camera.get_frame(timeout_ms=timeout)
+        return frame
+
+    def _get_buffer_and_ancillary_data(self, frame):
+        """Convert the frame and get the real exposure time.
+
+        Parameters
+        ----------
+        frame : Vimba.Frame
+            The frame from the camera.
+
+        Returns
+        -------
+        numpy.ndarray
```

Comment:
Again, since you are returning named variables, I'd do:

```
        buffer_array : `numpy.ndarray`
             The converted camera frame.
         actual_exp_time : `float`
             The actual frame exposure time (in seconds).
```

Also, note that I added the units of the `actual_exp_time` to the docstring.. please confirm that is really the case.

Reply:
Yes, the variable is in seconds. Changes made.